runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -946,6 +953,10 @@ private void process(
     final Windmill.WorkItem workItem = work.getWorkItem();
     final String computationId = computationState.getComputationId();
     final ByteString key = workItem.getKey();
+    if (work.isFailed()) {
+      LOG.debug("Not processing failed work for {}:\n{}", computationId, work);
+      return;
```

Comment:
I think just returning will leave things in a possibly odd state. I think that
computationState.completeWorkAndScheduleNextWorkForKey(
needs to be called at least.

Perhaps we could move this down to just where we would do the execution and throw the special exception then instead of executing. I don't think the work saved getting to that point is that expensive and that makes the error more common to existing processing errors.

Reply:
Done. Moved this into the try/catch block where processing happens.